ops/charm.py

```diff
@@ -261,62 +344,157 @@ class RelationCreatedEvent(RelationEvent):
+    """Event triggered when a relation is removed.
+
+    If a relation is being removed (`juju remove-relation` or `juju
+    remove-application`), once all the units have been removed, this
+    event will fire to signal that the relationship has been fully
+    terminated.
+
+    The event indicates that the current relation is no longer valid,
+    and that the charm’s software must be configured as though the
+    relation had never existed. It will only be called after every
+    callback method bound to :class: `RelationDepartedEvent` event has
+    been run. If a callback method bound to this event is being
+    executed, it is gauranteed that no remote units are currently
+    known locally.
     """
 
 
 class StorageEvent(HookEvent):
-    """Base class representing Storage related events."""
+    """Base class representing Storage related events.
+
+    Juju can provide a variety of storage types to a charms. The
+    charms can define several different types of storage that are
+    allocated from Juju. Changes in state of storage trigger sub-types
+    of :class: `StorageEvent` events.
+    """
 
 
 class StorageAttachedEvent(StorageEvent):
-    """Represents the `storage-attached` hook from Juju.
+    """Event triggered when new storage becomes available.
+
+    This event is triggered when new storage is available for the
+    charm to use.
 
-    Called when new storage is available for the charm to use.
+    Callback methods bound to this event allows the charm to run code
+    when storage has been added. Such methods will be run before the
+    :class: `InstallEvent` fires, so that the installation routine may
+    use the storage. The name prefix of this hook will depend on the
+    storage key defined in the `metadata.yaml` file.
     """
 
 
 class StorageDetachingEvent(StorageEvent):
-    """Represents the `storage-detaching` hook from Juju.
+    """Event triggered prior to removal of storage.
+
+    This event is triggered when storage a charm has been using is
+    going away.
 
-    Called when storage a charm has been using is going away.
+    Callback methods bound to this event allows the charm to run code
+    before storage is removed. Such methods will be run before storage
+    is detached, and always before the :class: `StopEvent` fires, thereby
+    allowing the charm to gracefully release resources before they are
+    removed and before the unit terminates. The name prefix of the
+    hook will depend on the storage key defined in the `metadata.yaml`
+    file.
     """
 
 
 class CharmEvents(ObjectEvents):
-    """The events that are generated by Juju in response to the lifecycle of an application."""
+    """Events generated by Juju pertaining to application lifecycle.
+
+    This class is used to create an event descriptor (`self.on`)
+    attribute for a Charm class that inherits from CharmBase. The
+    event descriptor may be used to setup event handlers for
```

Comment:
```suggestion
    event descriptor may be used to set up event handlers for
```
noun -> setup
verb -> set up